src/functions.jl

```diff
@@ -13,8 +13,119 @@ Interverts the car `i` with the car `j` in `solution.sequence`. Updates
 `solution.M1`, `solution.M2` and `solution.M3`.
 """
 function move_exchange!(solution::Solution, i::Int, j::Int, instance::Instance)
-    # TODO
-    return Solution
+    sol_i = solution.sequence[i]
+    solution.sequence[i] = solution.sequence[j]
+    solution.sequence[j] = sol_i
+    for k in 1:(instance.nb_HPRC + instance.nb_LPRC)
+        if k <= instance.nb_HPRC
+            if instance.HPRC_flag[i, k] == false && instance.HPRC_flag[j, k] == true
+                pos1 = i - ( instance.HPRC_q[k]-1)
+                for l in pos1:i
+                    solution.M1[k, l] += 1
+                end
+                pos2 = j - (instance.HPRC_q[k]-1)
+                for l in pos2:j
+                    solution.M1[k, l] -= 1
+                end
+                cpt1 = solution.M2[k, pos1-1]
+                cpt2 = solution.M3[k, pos1-1]
+                for l in pos1:solution.n
+                    if solution.M1[k, l] > instance.HPRC_p[k]
+                        cpt1 += 1
+                        solution.M2[k, l] = cpt1
+                    else
+                        solution.M2[k, l] = cpt1
+                    end
+                    if solution.M1[k, l] >= instance.HPRC_p[k]
+                        cpt2 += 1
+                        solution.M3[k, l] = cpt2
+                    else
+                        solution.M3[k, l] = cpt2
+                    end
+                end
+            end
+            if instance.HPRC_flag[i, k] == true && instance.HPRC_flag[j, k] == false
+               pos1 = i - (instance.HPRC_q[k] - 1)
+               for l in pos1:i
+                   solution.M1[k, l] -= 1
+               end
+               pos2 = j - (instance.HPRC_q[k]-1)
+               for l in pos2:j
+                   solution.M1[k, l] += 1
+               end
+               cpt1 = solution.M2[k, pos1-1]
+               cpt2 = solution.M3[k, pos1-1]
+               for l in pos1:solution.n
+                   if solution.M1[k, l] > instance.HPRC_p[k]
+                       cpt1 += 1
+                       solution.M2[k, l] = cpt1
+                   else
+                       solution.M2[k, l] = cpt1
+                   end
+                   if solution.M1[k, l] >= instance.HPRC_p[k]
+                       cpt2 += 1
+                       solution.M3[k, l] = cpt2
+                   else
+                       solution.M3[k, l] = cpt2
+                   end
+               end
+            end
+        else
+            if instance.LPRC_flag[i, k - instance.HPRC_q[k]] == false && instance.LPRC_flag[j, k - instance.HPRC_q[k]] == true
```

Comment:
`- instance.HPRC_q[k]` should have been `- instance.nb_HPRC`. Now we can have `instance.RC_flag[i, k]` instead of `instance.LPRC_flag[i, k - instance.nb_HPRC]`

In fact this whole part is not necessary anymore